File: src/main/resources/static/ts/GameObjects/Consumable.ts
import GameObject = require('./GameObject');
import PhysicsState = require('../PhysicsState');
import Rect = require('../Rect');

export = class Consumable extends GameObject {
    speedBoost: number;
    jumpSpeedBoost: number;

    constructor(parent: GameObject | null, frame: Rect, speedBoost: number, jumpSpeedBoost: number) {
        super(parent, frame);

        this.physics = new PhysicsState(this);

        this.speedBoost = speedBoost;
        this.jumpSpeedBoost = jumpSpeedBoost;
    };
}

File: src/main/kotlin/me/demoniacdeath/onlineplatformer/Consumable.kt
package me.demoniacdeath.onlineplatformer

class Consumable(
        frame: Rect,
        val consumablePowerSpeedBoost: Double,
        val consumablePowerJumpSpeedBoost: Double
) : GameObject(frame) {
    override val type = "Consumable"
    init {
        physics = PhysicsState(this)
    }
}
